src/component/note-item/note-item.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Modal from '../modal/modal';
+import NoteForm from '../note-form/note-form';
+
+export default class NoteItem extends React.Component {
+  render() {
+    const { note, handleDeleteNote, handleUpdateNote } = this.props;
+
+    const showModal = () => handleUpdateNote({ ...note, editing: true });
+
+    const hideModal = () => handleUpdateNote({ ...note, editing: false });
+
+    const updateAndClose = (updatedNote) => {
+      handleUpdateNote({ ...updatedNote, editing: false });
+    };
+
+    return (
+      <li className='note-item'>
+        <h3>{note.title}</h3>
+        <p>{note.content}</p>
+        <button id={note.id} onClick={handleDeleteNote}>Delete This Note</button>
```

Comment:
I like your use of the native `id` attribute in HTML to set your note id to the native id, but I've seen that this can lead to bugs and confusion when combining React props/state values with native attribute values. 

As long as you are aware and have a good handle on it (which you do in your project), I'm fine with you using it in future projects. However, when working in a team setting, I'd recommend toward finding another method of passing in the `id` that doesn't conflict with the native attributes.